x-pack/plugins/cases/server/connectors/cases/README.md

```diff
@@ -0,0 +1,577 @@
+# Summary
+
+
+The case action groups related alerts to cases automatically based on rules and conditions when alerts are detected. Specifically the case action:
+
+1. Allows users to automatically attach alerts to a case.
+2. Allows users to group alerts by a field and attach all alerts of each group to a case. Each group will be attached to its own case.
+3. Allows users to define a time window specifying when alerts will be attached to an existing case instead of creating a new one based on time.
+4. Allows users to configure if they want to reopen the case if it is closed.
+
+# Architecture  
+
+## Terminology
+
+|      |                                   |
+| :--- | :-------------------------------- |
+| GD   | Grouping definition               |
+| RGD  | Rule and grouping definition      |
+| RGDP | Rule and grouping definition pair |
+
+## Connector adapter
+
+## High-level flow
+
+The case action groups all alerts based on the grouping field defined by the user. Then the case action for each group does the following:
+1. Check if there is a case that already represents the specific group for the specific rule.
+2. If not, it will create a new case and attach the alerts to the new case. If there is a case, then it will do the following checks:
+   1. Check if the case has already 1K alerts attached to it. If yes it will log a warning and terminate.
+   2. Check if the case is older than the defined time window. If yes it will create a new case as described in Step 1.
+   3. Check if the case is closed. If it is, it will check if the case should be reopened. If yes the case action will reopen the case and attach the alerts to it. If not it will create a new case and attach the alerts to the new case.
+
+If an alert does not belong to a group it will not be attached to any case. Also, if no grouping field is configured by the user, all alerts will be attached to the same case.
+
+```mermaid
+flowchart TB
```

Comment:
oh wow. Didn't know how to attach flowchart to readme. 😃

Reply:
Yes, GitHub supports [Mermaid](https://mermaid.js.org/intro/getting-started.html) diagrams!